src/main/scala/is/hail/methods/KinshipMatrix.scala

```diff
@@ -11,7 +13,7 @@ import is.hail.utils._
 /**
   * Represents a KinshipMatrix. Entry (i, j) encodes the relatedness of the ith and jth samples in sampleIds.
   */
-class KinshipMatrix(val hc: HailContext, val matrix: IndexedRowMatrix, val sampleIds: Array[String]) {
+class KinshipMatrix(val hc: HailContext, val matrix: IndexedRowMatrix, val sampleIds: Array[String], val numVariantsUsed: Long) {
```

Comment:
What advantage is there to making it a case class? Most of the things we get for free from declaring as case are weird because HailContext is in the constructor. 

1. Pattern matching is weird with HailContext in the constructor, not clear when I'd ever pattern match on a KinshipMatrix anyway. 
2. Equality and hashcode methods on KinshipMatrix should not look at the HailContext. 
3. toString method should not look at HailContext.